presto-main/src/main/java/io/prestosql/operator/window/LastValueFunction.java

```diff
@@ -42,23 +54,30 @@ public void processRow(BlockBuilder output, int frameStart, int frameEnd, int cu
             return;
         }
 
-        int valuePosition = frameEnd;
-
+        int returnInd = frameEnd;
         if (ignoreNulls) {
-            while (valuePosition >= frameStart) {
-                if (!windowIndex.isNull(argumentChannel, valuePosition)) {
-                    break;
-                }
-
-                valuePosition--;
-            }
+            returnInd = searchWindowAndUpdateIndices(frameStart, frameEnd);
+        }
+        windowIndex.appendTo(argumentChannel, returnInd, output);
+    }
 
-            if (valuePosition < frameStart) {
-                output.appendNull();
```

Comment:
previously there was a case where we would `appendNull`

not sure yet under what conditions, but where is it now?